### PR TITLE
[BD-26] Create combined API endpoint for exam and attempts

### PR DESCRIPTION
**Pull request by UvgenGen**
[EDUCATOR-5733](https://openedx.atlassian.net/browse/EDUCATOR-5733) [BD-26] Create combined API endpoint for exam and attempts

### CODE REVIEW

File: edx_proctoring/views.py [hunk 1]
```diff
@@ -168,6 +170,53 @@ def handle_exception(self, exc):
         return resp
 
 
+class ProctoredExamAttemptsView(ProctoredAPIView):
```
Comment (Zacharis278) [inline edx_proctoring/views.py:173]: this class is very similar to StudentProctoredExamAttemptCollection. I think we should find a way to either make use of the existing view, or have both of these extend from a common base class.

File: edx_proctoring/api.py [hunk 2]
```diff
@@ -549,6 +554,77 @@ def get_exam_attempt_by_code(attempt_code):
     return _get_exam_attempt(exam_attempt_obj)
 
 
+def get_exam_attempt_data(exam_id, attempt_id, is_learning_mfe=False, request=None):
```
Comment (TamoshaytisV) [inline edx_proctoring/api.py:557]:
```suggestion
def get_exam_attempt_data(exam_id, attempt_id, override_exam_url=None):
```

File: edx_proctoring/api.py [hunk 3]
```diff
@@ -549,6 +554,77 @@ def get_exam_attempt_by_code(attempt_code):
...
+
+    low_threshold = int(low_threshold_pct * float(attempt['allowed_time_limit_mins']) * 60)
+    critically_low_threshold = int(
+        critically_low_threshold_pct * float(attempt['allowed_time_limit_mins']) * 60
+    )
+
+    # resolve the LMS url, note we can't assume we're running in
+    # a same process as the LMS
+    if is_learning_mfe:
+        usage_key = UsageKey.from_string(exam['content_id'])
```
Comment (TamoshaytisV) [inline edx_proctoring/api.py:584]: Move building url to a separate function or inside View

**Review by TamoshaytisV — CHANGES_REQUESTED** [review]

File: edx_proctoring/api.py [hunk 4]
```diff
@@ -549,6 +554,77 @@ def get_exam_attempt_by_code(attempt_code):
...
+        (
+            course_key, chapter, section, vertical_unused,
+            position, final_target_id
+        ) = path_to_location(modulestore(), usage_key, request)
+        exam_url_path = get_learning_mfe_courseware_url(course_key, section, final_target_id)
```
Comment (TamoshaytisV) [inline edx_proctoring/api.py:589]: @UvgenGen i saw that in platform this function is called `make_learning_mfe_courseware_url`

File: edx_proctoring/views.py (hunk 1 above)
Comment (TamoshaytisV) [reply edx_proctoring/views.py:173]: addressed

**Review by TamoshaytisV — APPROVED** [review]

**Pull request merged**